docs/docs/gatsby-lifecycle-apis.md

```diff
@@ -13,4 +13,49 @@ Gatsby's design principles include:
   trivial and encouraged.
 - Plugins are easy to open source and reuse. They're just NPM packages.
 
-See the links along the left under "REFERENCE" for the full API documentation.
+# High level Overview
+
+## Bootstrap sequence
+
+During "bootstrap" gatsby:
+
+- reads `gatsby-config.js` to load in your list of plugins
+- initializes its cache (stored in `/.cache`)
+- pulls in and preprocesses data ("source and transform nodes") into a GraphQL schema
+- creates pages in memory
+  - from your `/pages` folder 
+  - from your `gatsby-node.js` if you implement `createPages`/`createPagesStatefully` (e.g. templates)
+  - from any plugins that implement `createPages`/`createPagesStatefully`
+- extracts, runs, and replaces graphql queries for pages and `StaticQuery`s
+- writes out the pages to cache
+
+In development this is a running process powered by [Webpack](https://github.com/gatsbyjs/gatsby/blob/dd91b8dceb3b8a20820b15acae36529799217ae4/packages/gatsby/package.json#L128) and [react-hot-loader](https://github.com/gatsbyjs/gatsby/blob/dd91b8dceb3b8a20820b15acae36529799217ae4/packages/gatsby/package.json#L104), so changes to any files get re-run through the sequence again, with smart cache invalidation.
```

Comment:
We should also mention that source plugins like gatsby-source-filesystem can also be watching files for changes which can trigger re-running queries. Also that queries are watched so if you modify a query it's reloaded.